ts/features/bonus/bpd/screens/iban/IbanCTAEditScreen.tsx

```diff
@@ -112,6 +113,22 @@ const IbanCTAEditScreen: React.FC<Props> = (props: Props) => {
   );
 };
 
+/**
+ * Landing screen from the CTA message that asks to review user's IBAN insertion
+ */
+const IbanCTAEditScreen: React.FC<Props> = (props: Props) => {
+  const navigation = useNavigationContext();
+  if (!props.bpdRemoteConfig?.program_active) {
+    Alert.alert(
+      I18n.t("bonus.bpd.title"),
+      I18n.t("bonus.bpd.iban.bpdCompletedMessage")
+    );
+    navigation.goBack();
+    return null;
+  }
+  return <InnerIbanCTAEditScreen {...props} />;
```

Comment:
I see you are enforcing this pattern (which I like and I've used often in the past). 
I think it would be better to further isolate the `Inner*` components by defining separate `Props` since they normally don't need the entire set of `Props` defined for their containers.

Namely, I believe that we should start reducing the usage of state, dispatchers, context, ... to the bare minimum (aka none 😃 ). The entire point is to make the instantiation of the `Inner*` components trivial, thus enabling things like:

* extensive unit testing
* fuzzy testing
* mount the components into _showrooms_ or _playgrounds_ for presentation, testing, design purposes

Reply:
I agree, while in these cases the "split" of the component in `Inner*` or `Base*` was driven by the need to isolate conditional renders for hooks, I'm an huge fan of the composition and reuse of a common behaviour and separate the component logic from the databinding with the store.